core/src/main/java/org/apache/carbondata/scan/filter/FilterUtil.java

```diff
@@ -1401,8 +1401,7 @@ public static void logError(Throwable e, boolean invalidRowsPresent) {
   public static boolean nanSafeEqualsDoubles(Double d1, Double d2) {
     Boolean xIsNan = Double.isNaN(d1);
     Boolean yIsNan = Double.isNaN(d2);
-    if ((xIsNan && yIsNan) || (d1.doubleValue() == d2.doubleValue())) {
-
+    if ((d1.doubleValue() == d2.doubleValue()) || (xIsNan && yIsNan)) {
```

Comment:
Boolean xIsNan = Double.isNaN(d1); these variables are already computed, no use of moving them. Instead use ( Double.isNaN(d1) && Double.isNaN(d2))